Ignore pool order in `STAKE_POOLS_LIST_01`.

jonathanknowles opened the pull request:
# Issue Number

#1940 

# Overview

This PR changes the API `STAKE_POOLS_LIST_01` test to **ignore** the order in which pools are listed by the `ListStakePools` API operation.

It prevents spurious failures caused by pools being listed in unexpected orders.

jonathanknowles committed:
Allow for different stake pool orderings in `STAKE_POOLS_LIST_01`.

KtorZ commented on lib/core-integration/src/Test/Integration/Scenario/API/Shelley/StakePools.hs:
```diff
@@ -597,22 +597,24 @@ spec = do
+                        & either (error . show) Prelude.id
+                        & fmap (view #retirement)
+                        & fmap (fmap (getApiT . view #epochNumber))
+                        & Set.fromList
```
Suggestion: use `getFromResponse` and `#getApitT` instead:

```hs
retirements <- fmap (view (#epochNumber . #getApiT)) <$> getFromResponse #retirement response
```

jonathanknowles committed:
Use `#getApiT` to simplify test.

In response to review feedback:

https://github.com/input-output-hk/cardano-wallet/pull/1941#discussion_r459301140

paweljakubas reviewed (approved):
nice catch!

jonathanknowles committed:
Use function `getRetirementEpoch` to simplify test.

jonathanknowles committed:
Use `getFromResponse` to simplify test.

In response to review feedback:

https://github.com/input-output-hk/cardano-wallet/pull/1941#discussion_r459301140

KtorZ commented on lib/core-integration/src/Test/Integration/Scenario/API/Shelley/StakePools.hs:
```diff
@@ -597,22 +598,29 @@ spec = do
+                        view #retirement
+
+                let actualRetirementEpochs =
+                        getFromResponse Prelude.id response
```
by the way: getFromResponse expects a lens as a first argument, so `getFromResponse #retirement` works just fine.

jonathanknowles replied:
> by the way: getFromResponse expects a lens as a first argument, so `getFromResponse #retirement` works just fine.

We're dealing with a list here though, so the expression `getFromResponse #retirement response` produces the error:

```
The type [ApiStakePool] does not contain a field named 'retirement'
```

KtorZ replied:
Ah! It's a list of ApiStakePool, indeed.
